Dockerfile.management

```diff
@@ -0,0 +1,12 @@
+FROM registry.ci.openshift.org/openshift/release:golang-1.16 as builder
```

Comment:
I see the point of splitting out the CLI image, as we don't need the CLI in the image on the cluster and it will end up occupying quite a bit of space once we build it for multiple architectures and OSes. But I am not really sure what the advantage is of having a distinct operator image? It will break the possibility of passing a hypershift image to `hypershift install` that is then used for all hypershift components that are created for each cluster and thus worsen the UX for developing.

Reply:
I agree. Though for the image that belongs to the CVO payload (Dockerfile.control-plane) we can safely drop the hypershift-operator as in this PR, right? Then for dev purposes as you describe we could keep the cp binaries within the operator image.